.github/copilot-instructions.md

```diff
@@ -0,0 +1,188 @@
+  - Data ingestion, ID assignment, merging, indexing, linking
+  - Neo4j database creation
+  - Solr and SQLite exports
+- **webapp/**: Web application stack
+  - `grebi_api/`: Java Spring Boot REST API
+  - `grebi_ui/`: TypeScript/React frontend
+  - `grebi_metadata_service/`: Java Spring Boot metadata service
+  - `grebi_resolver_service/`: Java Spring Boot resolver service
+- **query_templates/**: YAML query templates for common queries
+- **materialised_queries/**: Cypher queries that are periodically executed
+- **configs/**: YAML configuration files for datasources and subgraphs
+
+## Technology Stack
+
+### Rust (dataload pipeline)
+- **Version**: 1.79+
+- **Build**: `cargo build --release`
+- **Style**: Follow standard Rust conventions (rustfmt)
+- **Dependencies**: Managed via Cargo.toml
+- **Key Libraries**: 
+  - serde_json for JSON processing
+  - clap for CLI argument parsing
+  - Custom grebi_shared library for common functionality
+
+### Java (Spring Boot services)
+- **Version**: Java 17+
+- **Build**: Maven (`mvn clean install`)
+- **Framework**: Spring Boot
+- **Style**: Follow standard Java conventions
+- **Package Structure**: `uk.ac.ebi.grebi.*`
+
+### TypeScript/React (UI)
+- **Framework**: React with Material-UI components
+- **Build**: npm/yarn
+- **Style**: Use functional components with hooks
+- **State Management**: React hooks (useState, useMemo, etc.)
+- **Syntax Highlighting**: Uses PrismJS for code display
+
+### Python (configuration scripts)
+- **Version**: Python 3.11
+- **Style**: PEP 8
+- **Usage**: Primarily for generating YAML configurations
+
+## Code Style Guidelines
+
+### General
+- Follow the existing code style in each language
+- Use descriptive variable names
+- Add comments for complex logic, but prefer self-documenting code
+- Keep functions small and focused
+
+### Rust
+- Use `snake_case` for functions and variables
+- Use `PascalCase` for types and traits
+- Prefer `Result` and `Option` over panicking
+- Use `eprintln!` for progress/debug messages to stderr
+- Stream processing: Read from stdin, write to stdout when possible
+
+### Java
+- Use `camelCase` for methods and variables
+- Use `PascalCase` for classes
+- Follow Spring Boot conventions for controllers, services, and repositories
+- Use `@Autowired` for dependency injection
+- Use SLF4J for logging
+
+### TypeScript/React
+- Use `camelCase` for variables and functions
+- Use `PascalCase` for components
+- Prefer functional components with hooks
+- Use TypeScript types/interfaces for props
+- Use Material-UI components consistently
+
+## Data Processing Patterns
+
+### Pipeline Architecture
+1. **00_fetch_data/**: Download raw data from external sources
+2. **01_ingest/**: Transform data into JSONL format and extract identifiers
+3. **02_assign_ids/**: Build equivalence groups and assign canonical IDs to nodes
+4. **03_merge/**: Merge nodes from different datasources based on equivalence groups (clique merging)
```

Comment:
merge properties about the same clique from multiple files into one json object per clique